CK3toEU4/Source/Mappers/LocalizationMapper/LocalizationMapper.cpp

```diff
@@ -43,52 +43,64 @@ void mappers::LocalizationMapper::scrapeLanguage(const std::string& language, co
 
 void mappers::LocalizationMapper::scrapeStream(std::istream& theStream, const std::string& language)
 {
+	std::string line;
+	std::getline(theStream, line); // Dropping first line and BOM with it.
+
 	while (!theStream.eof())
 	{
-		std::string line;
-		getline(theStream, line);
-
-		if (line[0] == '#' || line[1] == '#' || line.length() < 4)
-			continue;
-
-		const auto sepLoc = line.find_first_of(':');
-		if (sepLoc == std::string::npos)
-			continue;
-		const auto key = line.substr(1, sepLoc - 1);
-		const auto newLine = line.substr(sepLoc + 1, line.length());
-		const auto quoteLoc = newLine.find_first_of('\"');
-		const auto quote2Loc = newLine.find_last_of('\"');
-		if (quoteLoc == std::string::npos || quote2Loc == std::string::npos || quote2Loc - quoteLoc == 0)
-			continue;
-		const auto value = newLine.substr(quoteLoc + 1, quote2Loc - quoteLoc - 1);
-
-		if (localizations.count(key))
+		std::getline(theStream, line);
+		const auto [key, value] = determineKeyLocalizationPair(line);
+		if (!key.empty() && !value.empty())
 		{
-			if (language == "english")
-				localizations[key].english = value;
-			if (language == "french")
-				localizations[key].french = value;
-			if (language == "german")
-				localizations[key].german = value;
-			if (language == "spanish")
-				localizations[key].spanish = value;
-		}
-		else
-		{
-			LocBlock newBlock;
-			if (language == "english")
-				newBlock.english = value;
-			if (language == "french")
-				newBlock.french = value;
-			if (language == "german")
-				newBlock.german = value;
-			if (language == "spanish")
-				newBlock.spanish = value;
-			localizations.insert(std::pair(key, newBlock));
+			if (localizations.count(key))
+			{
+				if (language == "english")
+					localizations[key].english = value;
```

Comment:
```suggestion
					localizations.at(key).english = value;
```
at() is safer because it does bounds checking.